src/ast/analysis/Type.h

```diff
@@ -49,18 +59,44 @@ class TypeAnalysis : public Analysis {
      * a set of potential types. If the set associated to an argument is empty,
      * no consistent typing can be found and the rule can not be properly typed.
      *
-     * @param env a typing environment describing the set of available types
-     * @param clause the clause to be typed
-     * @param program the program
-     * @return a map mapping each contained argument to a a set of types
+     * @return a map mapping each contained argument to a set of types
      */
     static std::map<const Argument*, TypeSet> analyseTypes(
-            const TranslationUnit&, const Clause&, std::ostream* /*logs*/ = nullptr);
+            const TranslationUnit& tu, const Clause& clause, std::ostream* logs = nullptr);
+
+    // Checks whether an argument has been assigned a valid type
+    bool hasValidTypeInfo(const Argument* argument) const;
+
+    /** -- Functor-related methods -- */
+    bool hasProcessedFunctor(const Functor* functor) const;
+    IntrinsicFunctors validOverloads(const ast::IntrinsicFunctor& func) const;
```

Comment:
Parameter names are ideally meaningful, but should be at least consistent. Elsewhere you use `inf` for an IntrinsicFunctor, so maybe something related to that?